cmake/CMakeLists.txt

```
@@ -468,6 +488,7 @@ set(CMAKE_TUNE_FLAGS "${CMAKE_TUNE_DEFAULT}" CACHE STRING "Compiler and machine
 separate_arguments(CMAKE_TUNE_FLAGS)
 foreach(_FLAG ${CMAKE_TUNE_FLAGS})
   target_compile_options(lammps PRIVATE ${_FLAG})
```

I think it is a bit cumbersome to have to add the flags in all these 3+ places. I think we could achieve the same by:
```suggestion
  target_compile_options(lammps PUBLIC $<BUILD_INTERFACE:${_FLAG}>)
```
here and then the `lammps` consumers such as `lmp`, unittest` would get them automatically.

Another options would be to create a `flags` target, e.g.:
```
add_library(LAMMPS::flags INTERFACE IMPORTED)
target_compile_options(LAMMPS::flags PUBLIC $<BUILD_INTERFACE:${_FLAG}>)
```
and then use `target_link_libraries(<foo> PRIVATE LAMMPS::flags)` to inject the flags for each target.
That would also work for `covals`, which isn't linked against the `lammps` target. With the `flags` target we could clean up the `LAMMPS_SIZE` defines all over the place, too.